packages/uploadthing/src/internal/types.ts

```diff
@@ -97,13 +97,13 @@ type UploadErrorFn<TArgs extends AdapterFnArgs<any, any, any>> = (
 ) => MaybePromise<void>;
 
 export interface UploadBuilder<TParams extends AnyParams> {
-  input: <TParser extends JsonParser>(
+  input: <TIn extends Json, TOut>(
```

Comment:
only the input needs to be json - using transformations inside the schema should be valid as that data doesn't move over the network